app/scripts/controllers/replicaSet.js

```diff
@@ -0,0 +1,448 @@
+            updateDC(deployment);
+            updateHPAWarnings();
+
+            // If we found the item successfully, watch for changes on it
+            watches.push(DataService.watchObject($scope.resource, $routeParams.replicaSet, context, function(deployment, action) {
+              if (action === "DELETED") {
+                $scope.alerts["deleted"] = {
+                  type: "warning",
+                  message: "This " + displayKind + " has been deleted."
+                };
+              }
+              $scope.deployment = deployment;
+
+              if (!$scope.forms.envForm || $scope.forms.envForm.$pristine) { 
+                copyDeploymentAndEnsureEnv(deployment);
+              } else {
+                $scope.alerts["background_update"] = {
+                  type: "warning",
+                  message: "This " + displayKind + " has been updated in the background. Saving your changes may create a conflict or cause loss of data.",
+                  links: [
+                    {
+                      label: 'Reload environment variables',
+                      onClick: function() {
+                        $scope.clearEnvVarUpdates();
+                        return true;
+                      }
+                    }
+                  ]
+                };
+              }
+
+              setLogVars(deployment);
+              updateHPAWarnings();
+            }));
+
+            if ($scope.deploymentConfigName) {
+              // Check if we're the active deployment to enable or disable scaling.
+              watchActiveDeployment();
+            }
+
+            $scope.$watch('deployment.spec.selector', function() {
+              selector = new LabelSelector($scope.deployment.spec.selector);
+              updatePodsForDeployment();
+            }, true);
+
+            watches.push(DataService.watch("pods", context, function(podData) {
+              pods = podData.by('metadata.name');
+              updatePodsForDeployment();
+            }));
+          },
+          // failure
+          function(e) {
+            $scope.loaded = true;
+            $scope.alerts["load"] = {
+              type: "error",
+              message: "The " + kind + " details could not be loaded.",
+              details: "Reason: " + $filter('getErrorDetails')(e)
+            };
+          }
+        );
+
+        function extractPodTemplates() {
+          angular.forEach($scope.deployments, function(deployment, deploymentId){
+            $scope.podTemplates[deploymentId] = deployment.spec.template;
+          });
+        }
+
+        watches.push(DataService.watch($scope.resource, context, function(deployments, action, deployment) {
+          $scope.deployments = deployments.by("metadata.name");
+          extractPodTemplates();
+          ImageStreamResolver.fetchReferencedImageStreamImages($scope.podTemplates, $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, context);
```

Comment:
related to the previous comment, this really should be in the get and watchObject and not here since we only need the pod template image data for the main RC/RS